src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -181,6 +180,12 @@ impl FlowServiceImpl {
     ) -> Result<(), InternalError> {
         match &flow_key {
             FlowKey::Dataset(dataset_flow_key) => {
+                self.state
+                    .lock()
+                    .unwrap()
+                    .active_configs
+                    .add_dataset_flow_config(dataset_flow_key, rule.clone());
+
```

Comment:
This object's purpose was to work with automatic triggers only (keep only configs of enabled flow keys). You are giving it a different sense of a setting cache. But compacting is always manuall, and you can request the current config there instead of this.

Reply:
I just move it to save configuration before event initialisation, due to now we are storing this snapshot in the Initiated event